### ISSUE TITLE
IIPImageServer does not work with the latest OSD release v2.2.1 

### ISSUE THREAD

**Issue by johnico** [openseadragon/openseadragon#1195]
Hi
I have install iipimage with fastcgi  in local server (wamp,windows) 
from this site: http://iipimage.sourceforge.net/

and the server looks like it works,
there is a working example here : http://merovingio.c2rmf.cnrs.fr/iipimage/openseadragon/
and i m trying this code: 
    var viewer = OpenSeadragon({
        id: "openseadragon",
        prefixUrl: "/",
        tileSources: "/fcgi-bin/iipsrv.fcgi?Deepzoom=test.tif"
    });

the test.tif is a file i downloaded from here: http://merovingio.c2rmf.cnrs.fr/iipimage/PalaisDuLouvre.tif
i got this error from the server

Error/12:2 2 Deepzoom

is any tutorial for seadrgon and fastcgi ? or any other examples?
thanks

**Comment by iangilman** [comment]
You're probably better off asking this question to the IIPImage team: 

http://iipimage.sourceforge.net/contact/

...but I'll tag some of the folks here who have some IIPImage experience, in case they've got something to add: @jpstroop @azaroth42 @boskar

**Comment by jpstroop** [comment]
@johnico @iangilman : I think it's @ruven you want. Most of the people you listed are IIIFers.

**Comment by ruven** [comment]
Your tile source should be:

`"/fcgi-bin/iipsrv.fcgi?DeepZoom=test.tif.dzi"`

See here for more details: [http://iipimage.sourceforge.net/documentation/clients/#client](http://iipimage.sourceforge.net/documentation/clients/#client)

**Comment by comrade2sun** [comment]
Just  two quick hints:

➝ IIPImage does work for me on local server (apache) with
```
//! OpenSeadragon 1.2.1
//! Built on 2015-01-29
//! Git commit: v1.2.1-0-2f9563d
```

it does not work with the latest version

> //! openseadragon 2.2.0
> //! Built on 2016-05-26
> //! Git commit: v2.2.0-0-373e61e

the version I am using regularly for some other stuff.

➝ You are missing the appendix 'dzi' as ruven already pointed out

**Comment by johnico** [comment]
@comrade2sun . Could you share your code + a dzi file example  ?

**Comment by comrade2sun** [comment]
The basic setup is like this:

```js
    var viewer = OpenSeadragon({
        id: "openseadragon",
        prefixUrl: "icons/",     //navigation-icons
	tileSources: "/fcg-bin/iipsrv.fcgi?Deepzoom=123test.tif.dzi"   
    });
```
   
It's not necessary for you to create a dzi-file like '123test.dzi', as IIPImageServer creates the dzi on the fly. 
As I mentioned in my first comment, IIPImageServer does not work with the latest OSD release v2.2.1 and release v2.2.0, but with v2.0.0 and v1.2.1

Therefore, you should try <a href="https://github.com/openseadragon/openseadragon/releases/tag/v2.0.0">Release v2.0.0 </a> if this does solve your issue. 

There might be other causes  as well:
● incorrect directory options in your `http-iipimage.conf`
● faulty pyramidal TIFF file  _(I recommend to create your TIFF with VIPS)_
● `fcg-bin` is black listed by an adblocker like Adblock Plus _(simply rename your folder in `iipserver-bin` or similar)_

Good Luck

**Comment by johnico** [comment]
Hi 
Found a nice solution for version 2.2.0 
I guess there is a bug with the OSD , probably its prase the url from the tilesource and remove the params after ?Deepzoom.. 

I have fixed it by adding a rule to .htaccess:

`RewriteRule ^folder/(.*)$ "/fcgi-bin/iipsrv.fcgi?Deepzoom=$1" [L]
`
and change the call from  the OSD  viewer object to : 
```
 var tileSource = { tileSource : 'dzi/'+"full_path_to_tif.dzi file "};
viewer.open(tilesource)

```

**Comment by iangilman** [comment]
@jpstroop Thanks for pointing me in the right direction, and apologies for tagging the wrong people... 

@comrade2sun Can you say more about the issue with 2.2.1? Any thoughts on how to fix it?

**Comment by comrade2sun** [comment]
@iangilman
_in nuce_
This bug looks similar like this one

<a href="https://github.com/openseadragon/openseadragon/issues/1017">IIPImage server, dzi, osd version 1.2.1 to 2.2.1 · Issue #1017 ·</a>

The difference is, that I am using a "Tiled Multi-Resolution TIFF" converted with "nip2-7.26.3"  and not a "JPEG2000". 

The DZI - a file named iipsrv.fcgi - created by IIPImage Server reads like this:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Image xmlns="http://schemas.microsoft.com/deepzoom/2008"
TileSize="256" Overlap="0" Format="jpg"><Size Width="7235" Height="5953"/>
</Image>
```

Currently  I am running (OS WIN 10) on localhost  XAMPP 1.7.7  and IIPImage Server. Version 0.9.9

What caused this issue, must be in the changes from  release v2.1.0 to release v2.2.0 and is maybe again the  regex as mentioned by <a href="https://github.com/openseadragon/openseadragon/issues/1017#issuecomment-243417998">librarymark</a> and the related discussions.

I dont know, if this info is much of a help ...

**Comment by iangilman** [comment]
@comrade2sun Great, yes, that is helpful! Looks like #1017 is marked as fixed... could you try building the latest master and verifying that it works with IIP?

**Comment by comrade2sun** [comment]
@iangilman 
In the assumption this is the fix you are talking about
<a href="https://github.com/openseadragon/openseadragon/pull/1074/commits/870bb2a99a815c214c07d7c16cbaaef9ee1dc5d6">See #832, further fix for dzitilesource guessing of tilesUrl, for iii… by altert · Pull Request #1074 · </a>

I replaced in openseadragon.js` line 11695`

```js
if (url && !options.tilesUrl) {
	options.tilesUrl = url.replace( 
	/* current regex - not working */
		/([^\/]+?)(\.(dzi|xml|js))?\/?(\?.*)?$/, '$1_files/'
	/* fix · Pull Request #1074 · - not working*/
		/([^\/]+?)(\.(dzi|xml|js))?(\?[^\/]*)?\/?$/, '$1_files/'
	/* old regex - OK */
		/([^\/]+)\.(dzi|xml|js)(\?.*|$)/, '$1_files/'
			);
```

This is what I read in the browser console (example)

> GET http://localhost/iipsrv-bin/iipsrv.fcgi_files/13/17_13.jpg

instead of 

> GET http://localhost/iipsrv-bin/iipsrv.fcgi 


... and this in the IIPImageServer.log:
```
Full Request is Deepzoom=123.tif.dzi
[1/1]: Command / Argument is Deepzoom : 123.tif.dzi
```

instead of
```
Full Request is Deepzoom=123.tif_files/13/17_13.jpg
[1/1]: Command / Argument is Deepzoom : 123.tif_files/13/17_13.jpg
```

Unless I made no mistake or typo, this fix does not work in this testcase ...

**Comment by iangilman** [comment]
@comrade2sun Okay, good to know! Do you have any thoughts about how to improve the regex so it supports IIP but also the other required patterns? I figure the first step is to define a new test for the IIP pattern (just like #1074 added new tests) and then work with the regex to satisfy all the tests. 

Keep in mind that the fix doesn't need to be a single regex... one option would be to try one regex and if it fails to make a match, try another.

**Comment by altert** [comment]
I've made a PR that should fix it. Could you test it on your setups?

**Comment by iangilman** [comment]
Fixed by #1199. Thank you @altert!
